gremlin-go/driver/connection_test.go

```diff
@@ -203,9 +196,35 @@ func TestConnection(t *testing.T) {
 			assert.NotNil(t, resultSet)
 			result := resultSet.one()
 			assert.NotNil(t, result)
-			assert.Equal(t, "[0]", result.GetString())
+			assert.Equal(t, "0", result.GetString())
 			err = client.Close()
 			assert.Nil(t, err)
 		}
 	})
+
+	t.Run("Test DriverRemoteConnection Next and HasNext", func(t *testing.T) {
+		if runIntegration {
+			// Add data
+			remote, err := NewDriverRemoteConnection(testHost, testPort)
+			assert.Nil(t, err)
+			assert.NotNil(t, remote)
+			g := Traversal_().WithRemote(remote)
+			dropGraph(t, g)
+			addTestData(t, g)
+
+			// Run traversal and test Next/HasNext calls
+			traversal := g.V().HasLabel(personLabel).Properties(nameKey)
```

Comment:
if the traversal was 
`g.V().HasLabel("invalid_label")`
should HasNext return false?

Current behaviour:
returns `true` since the channel would still be open.

Reply:
Should return false, since len(channel) should be 0

Reply:
The check for isEmpty() includes the a check to see if the channel is closed. Although the len  is 0 it would return true.

```
func (channelResultSet *channelResultSet) IsEmpty() bool {
	return channelResultSet.closed && len(channelResultSet.channel) == 0
}
```

Reply:
We could make it an OR clause, would that cause any colateral issues?

Reply:
AN-1051

Reply:
That could cause colateral issues will fix in other ticket